apps/rush-lib/src/cli/actions/ChangeAction.ts

```diff
@@ -100,32 +108,192 @@ export class ChangeAction extends BaseRushAction {
+    const sortedProjectList: string[] = this._getChangedPackageNames().sort();
+    if (sortedProjectList.length === 0) {
       this._logNoChangeFileRequired();
       this._warnUncommittedChanges();
       return Promise.resolve();
     }
 
-    this._prompt = inquirer.createPromptModule();
-    this._changeFileData = new Map<string, IChangeFile>();
-    this._changeComments = ChangeFiles.getChangeComments(this._getChangeFiles());
+    this._warnUncommittedChanges();
+
+    let changeFileDataPromise: Promise<Map<string, IChangeFile>>;
+    let allowOverwriteHandler: (filePath: string) => Promise<boolean>;
+    if (this._bulkChangeParameter.value) {
+      if (
+        !this._bulkChangeBumpTypeParameter.value ||
+        (
+          !this._bulkChangeMessageParameter.value &&
+          this._bulkChangeBumpTypeParameter.value !== ChangeType[ChangeType.none]
+        )
+      ) {
+        throw new Error(
+          `The ${this._bulkChangeBumpTypeParameter.longName} and ${this._bulkChangeMessageParameter.longName} ` +
+          `parameters must provided if the ${this._bulkChangeParameter.longName} flag is provided. If the value ` +
+          `"${ChangeType[ChangeType.none]}" is provided to the ${this._bulkChangeBumpTypeParameter.longName} ` +
+          `patameter, the ${this._bulkChangeMessageParameter.longName} parameter may be omitted.`
+        );
+      }
+
+      const email: string | undefined = this._changeEmailParameter.value || this._detectEmail();
+      if (!email) {
+        throw new Error(
+          'Unable to detect git email and an email address wasn\'t provided on the ' +
+          `${this._changeEmailParameter.longName} paramter.`
+        );
+      }
 
-    return this._promptLoop()
-      .catch((error: Error) => {
-        throw new Error(`There was an error creating a change file: ${error.toString()}`);
+      const errors: string[] = [];
+
+      const comment: string = this._bulkChangeMessageParameter.value || '';
+      const changeType: string = this._bulkChangeBumpTypeParameter.value;
+      const changeFileData: Map<string, IChangeFile> = new Map<string, IChangeFile>();
+      for (const packageName of sortedProjectList) {
+        const allowedBumpTypes: string[] = Object.keys(this._getBumpOptions(packageName));
+        let projectChangeType: string = changeType;
+        if (allowedBumpTypes.length === 0) {
+          projectChangeType = ChangeType[ChangeType.none];
+        } else if (
+          projectChangeType !== ChangeType[ChangeType.none] &&
+          allowedBumpTypes.indexOf(projectChangeType) === -1
+        ) {
+          errors.push(`The "${projectChangeType}" change type is not allowed for package "${packageName}".`);
+        }
+
+        changeFileData.set(
+          packageName,
+          {
+            changes: [
+              {
+                comment,
+                changeType: projectChangeType,
+                packageName
+              } as unknown as IChangeInfo
```

Comment:
`unknown as IChangeInfo` seems incorrect here.

The `IChangeInfo` declares `changeType` as being the enum number, not the string.  Also, isn't `ChangeFile` the thing that's supposed to build and save these files?  Looks like `IChangeInfo` is meant to be its API input, not its JSON file format.

Reply:
Good catch. This should be 'type'. Enum field changeType is parsed from the string field 'type' in the json file.